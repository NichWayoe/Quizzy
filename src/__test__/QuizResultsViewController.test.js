import QuizResultsViewController from '../Controllers/QuizResultsViewController.react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Quiz from '../Model/Quiz.react';

  describe('Test for QuizResultsViewController', () => {
    let answers;
    let userAnswers;
  
    beforeAll(() => {
      answers = {1:2, 2:2, 3:1}
      userAnswers = {1:2, 2:3, 3:1}
    });  
    test('checks for correct score displayed', () => {
    render(<QuizResultsViewController answers={answers} userAnswers={userAnswers}/>)
    expect(screen.queryByText(/2/))
    });
   
    it('renders correctly when there are quiz given items', () => {
                    const tree = renderer.create(<QuizResultsViewController answers={answers} userAnswers={userAnswers}/>).toJSON();
                    expect(tree).toMatchSnapshot();
                  });

    it('renders correctly when there are no items', () => {
                    const tree = renderer.create(<QuizResultsViewController/>).toJSON();
                    expect(tree).toMatchSnapshot();
                  });
  });