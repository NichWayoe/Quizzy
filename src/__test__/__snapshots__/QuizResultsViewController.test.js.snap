// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for QuizResultsViewController renders correctly when there are no items 1`] = `
<div>
  <div
    className="MuiBox-root css-ahbfdj"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root ForwardRef(Card)-root-2 css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
      >
        <div
          className="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
        >
          <img
            src="icon.png"
            style={
              Object {
                "paddingLeft": "25%",
              }
            }
          />
        </div>
        <div
          className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
        >
          <h3
            className="MuiTypography-root MuiTypography-h3 css-1m6imez-MuiTypography-root"
          >
            Your Score!
          </h3>
        </div>
      </div>
      <h1
        className="MuiTypography-root MuiTypography-h1 css-gq6uzy-MuiTypography-root"
      >
        0
      </h1>
    </div>
     
  </div>
</div>
`;

exports[`Test for QuizResultsViewController renders correctly when there are quiz given items 1`] = `
<div>
  <div
    className="MuiBox-root css-ahbfdj"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root ForwardRef(Card)-root-2 css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
      >
        <div
          className="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
        >
          <img
            src="icon.png"
            style={
              Object {
                "paddingLeft": "25%",
              }
            }
          />
        </div>
        <div
          className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
        >
          <h3
            className="MuiTypography-root MuiTypography-h3 css-1m6imez-MuiTypography-root"
          >
            Your Score!
          </h3>
        </div>
      </div>
      <h1
        className="MuiTypography-root MuiTypography-h1 css-gq6uzy-MuiTypography-root"
      >
        0
      </h1>
    </div>
     
  </div>
</div>
`;
